src/d/actor/d_a_obj_iceisland.cpp

```diff
@@ -5,95 +5,272 @@
 
 #include "d/actor/d_a_obj_iceisland.h"
 #include "d/d_procname.h"
+#include "d/d_com_inf_game.h"
+#include "d/res/res_gicel.h"
+#include "JSystem/JKernel/JKRExpHeap.h"
 
 /* 00000078-00000098       .text CheckCreateHeap__FP10fopAc_ac_c */
-static BOOL CheckCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL CheckCreateHeap(fopAc_ac_c* i_this) {
+    return static_cast<daObjIceisland_c*>(i_this)->CreateHeap();
 }
 
 /* 00000098-00000338       .text CreateHeap__16daObjIceisland_cFv */
-void daObjIceisland_c::CreateHeap() {
-    /* Nonmatching */
+BOOL daObjIceisland_c::CreateHeap() {
+    mDoExt_getGameHeap()->check();
+    
+    J3DModelData* model_data = (J3DModelData*)(dComIfG_getObjectRes("GiceL", GICEL_BDL_GICEL00));
+    JUT_ASSERT(0x66, model_data != NULL);
+    mpModel = mDoExt_J3DModel__create(model_data, 0, 0x11020203);
+
+    J3DAnmTextureSRTKey* btk1 = static_cast<J3DAnmTextureSRTKey*>(dComIfG_getObjectRes("GiceL", GICEL_BTK_GICEL00_01));
+    JUT_ASSERT(0x6D, btk1 != NULL);
+    int result1 = mBtkAnm1.init(model_data, btk1, true, J3DFrameCtrl::EMode_LOOP, 1.0f, 0, -1, false, 0);
+
+    J3DAnmTextureSRTKey* btk2 = static_cast<J3DAnmTextureSRTKey*>(dComIfG_getObjectRes("GiceL", GICEL_BTK_GICEL00_02));
+    JUT_ASSERT(0x73, btk2 != NULL);
+    int result2 = mBtkAnm2.init(model_data, btk2, true, J3DFrameCtrl::EMode_LOOP, 1.0f, 0, -1, false, 0);
+
+    J3DAnmTevRegKey * brk = static_cast<J3DAnmTevRegKey*>(dComIfG_getObjectRes("GiceL", GICEL_BRK_GICEL00));
+    JUT_ASSERT(0x7A, brk != NULL);
+    int result3 = mBrkAnm.init(model_data, brk, true, J3DFrameCtrl::EMode_NONE, 1.0f, 0, -1, false, 0);
+    return((mpModel != NULL) && (result1 != 0) && (result2 != 0) && (result3 != 0));  
 }
 
 /* 00000338-00000410       .text daObjIceisland_particle_set__16daObjIceisland_cFv */
 void daObjIceisland_c::daObjIceisland_particle_set() {
-    /* Nonmatching */
+    if(mEmitter1 == NULL) {
+        mEmitter1 = dComIfGp_particle_set(dPa_name::ID_SCENE_81AA, &current.pos, &current.angle);
+    }
+    if(mEmitter2 == NULL) {
+        mEmitter2 = dComIfGp_particle_set(dPa_name::ID_SCENE_81AB, &current.pos, &current.angle);
+    }
 }
 
 /* 00000410-00000588       .text CreateInit__16daObjIceisland_cFv */
 void daObjIceisland_c::CreateInit() {
-    /* Nonmatching */
+    cullMtx = mpModel->getBaseTRMtx();
+    cullSizeFar = 1.0f;
+    set_mtx();
+    dKy_tevstr_init(&mTevStr, fopAcM_GetHomeRoomNo(this), 0xFF);
+    g_env_light.settingTevStruct(TEV_TYPE_BG0, &current.pos, &mTevStr);
+    mEmitter1 = NULL;
+    mEmitter2 = NULL;
+    u8 switchNo = daObjIceisland_prm::getSwitchNo(this);
+    if(fopAcM_isSwitch(this, switchNo)){
+        mBrkAnm.setFrame(mBrkAnm.getEndFrame());
+        if(dComIfGs_getStartPoint() == 2 && current.roomNo == dComIfGs_getRestartRoomNo()){// some if
```

Comment:
Looks like an old comment
```suggestion
        if(dComIfGs_getStartPoint() == 2 && current.roomNo == dComIfGs_getRestartRoomNo()){
```